src/erlcloud_ec2.erl

```diff
@@ -2936,6 +2946,221 @@ net_if_params(#ec2_net_if{}=X) ->
+describe_flow_logs() ->
+    describe_flow_logs([], [], default_config()).
+
+-spec describe_flow_logs(aws_config()) -> ok_error(proplist());
+                        (ec2_flow_ids()) -> ok_error(proplist());
+                        (filter_list()) -> ok_error(proplist());
+                        (ec2_max_result()) -> ok_error(proplist(), ec2_token()).
+describe_flow_logs(Config)
+    when is_record(Config, aws_config) ->
+    describe_flow_logs([], Config);
+describe_flow_logs([{_, _} | _] = Filter) ->
+    describe_flow_logs([], Filter, default_config());
+describe_flow_logs(FlowIDs)
+    when is_list(FlowIDs) ->
+    describe_flow_logs(FlowIDs, [], default_config());
+describe_flow_logs(MaxResults)
+    when is_integer(MaxResults) ->
+    describe_flow_logs(MaxResults, default_config()).
+
+-spec describe_flow_logs(ec2_flow_ids(), filter_list()) -> ok_error(proplist());
+                        (ec2_flow_ids(), aws_config()) -> ok_error(proplist());
+                        (filter_list(), aws_config()) -> ok_error(proplist());
+                        (filter_list(), ec2_max_result()) -> ok_error(proplist());
+                        (ec2_max_result(), aws_config()) -> ok_error(proplist(), ec2_token());
+                        (ec2_max_result(), ec2_token()) -> ok_error(proplist(), ec2_token()). 
+describe_flow_logs(FlowIDs, none = Filter) ->
+    describe_flow_logs(FlowIDs, Filter, default_config());
+describe_flow_logs(FlowIDs, [{_, _} | _] = Filter)
+    when is_list(FlowIDs) ->
+    describe_flow_logs(FlowIDs, Filter, default_config());
+describe_flow_logs([{_, _} | _] = Filter, Config)
+    when is_record(Config, aws_config) ->
+    describe_flow_logs([], Filter, Config);
+describe_flow_logs([{_, _} | _] = Filter, MaxResults)
+    when is_integer(MaxResults) andalso MaxResults >= ?FLOWS_MR_MIN andalso MaxResults =< ?FLOWS_MR_MAX ->
+    describe_flow_logs(Filter, MaxResults, default_config());
+describe_flow_logs(FlowIDs, Config)
+    when is_list(FlowIDs), is_record(Config, aws_config) ->
+    describe_flow_logs(FlowIDs, [], Config);
+describe_flow_logs(MaxResults, Config)
+    when is_integer(MaxResults) andalso MaxResults >= ?FLOWS_MR_MIN andalso MaxResults =< ?FLOWS_MR_MAX,
+    is_record(Config, aws_config) ->
+    describe_flow_logs([], MaxResults, Config);
+describe_flow_logs(MaxResults, NextToken)
+    when is_integer(MaxResults) andalso MaxResults >= ?FLOWS_MR_MIN andalso MaxResults =< ?FLOWS_MR_MAX,
+    is_list(NextToken) ->
+    describe_flow_logs(MaxResults, NextToken, default_config()).
+
+-spec describe_flow_logs(ec2_flow_ids(), filter_list(), aws_config()) -> ok_error(proplist());
+                        (ec2_flow_ids(), filter_list(), ec2_max_result()) -> ok_error(proplist());
+                        (filter_list(), ec2_max_result(), aws_config()) -> ok_error(proplist(), ec2_token());
+                        (ec2_max_result(), ec2_token(), aws_config()) -> ok_error(proplist(), ec2_token()).
```

Comment:
I would also like to see interface `(Filter, MaxResults, NextToken)`, so that filtered paginated calls can be made using default config.